File: src/types/issue.ts
import { User } from './user';

export type Issue = {
  total: number;
  issues: SprintIssue[];
};

export type SprintIssue = {
  id: string;
  key: string;
  fields: SprintIssueFields;
};

type SprintIssueFields = {
  assignee?: User;
  customfield_10004?: number;
  summary: string;
  status: SprintIssueStatus;
};

type SprintIssueStatus = {
  id: string;
  name: string;
};

File: src-tauri/src/models/issue.rs
use crate::models::user::User;
use serde::{Deserialize, Serialize};

#[derive(Debug, Deserialize, Serialize)]
pub struct Issue {
    total: u32,
    issues: Vec<SprintIssue>,
}

#[derive(Debug, Deserialize, Serialize)]
pub struct SprintIssue {
    id: String,
    key: String,
    fields: SprintIssueFields,
}

#[derive(Debug, Deserialize, Serialize)]
pub struct SprintIssueFields {
    assignee: Option<User>,
    customfield_10004: Option<f32>,
    summary: String,
    status: SprintIssueStatus,
}

#[derive(Debug, Deserialize, Serialize)]
pub struct SprintIssueStatus {
    name: String,
}
